overlord/snapstate/snapstate.go

```diff
@@ -1909,6 +1917,16 @@ func Remove(st *state.State, name string, revision snap.Revision, flags *RemoveF
 		if tp, _ := snapst.Type(); tp == snap.TypeApp && removeAll {
 			ts, err := AutomaticSnapshot(st, name)
 			if err == nil {
+				if sz, err := EstimateSnapshotSize(st, name); err == nil {
+					requiredSpace := requiredSpaceWithMargin(uint64(sz))
+					if err := osutilCheckFreeSpace(dirs.SnapdStateDir(dirs.GlobalRootDir), requiredSpace); err != nil {
+						if _, ok := err.(*osutil.NotEnoughDiskSpaceError); ok {
+							return nil, fmt.Errorf("cannot create automatic snapshot when removing last revision of the snap: %v", err)
```

Comment:
I wonder if we need something extra here. Given that removing a snap maybe something you want to do to reclaim diskspace refusing to remove a snap because there is not enough disk-space is slightly confusing. So I wonder if at some higher layer we should hint at "snap remove --purge" or something?

Reply:
I think we want dedicated errors, so we can introduce  an error kind like "insufficient-space", then the cmd/snap code can print a reference to --purge based on that

Reply:
Added a dedicated error and will propose api/client side in a followup.

Reply:
Thank you!